src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -253,6 +253,66 @@ LedgerMaster::addHeldTransaction (
     mHeldTransactions.insert (transaction->getSTransaction ());
 }
 
+// Validate a ledger's close time and sequence number if we're considering
+// jumping to that ledger. This helps defend agains some rare hostile or
+// insane majority scenarios.
+bool
+LedgerMaster::canBeCurrent (std::shared_ptr<Ledger const> const& ledger)
+{
+    assert (ledger);
+
+    // Never jump to a candidate ledger that precedes our
+    // last validated ledger
+
+    auto validLedger = getValidatedLedger();
```

Comment:
Without a validated ledger, we accept this ledger as our new last fully-validated ledger? If so, consider returning true here.